File: hackerRank/algorithms/TheHurdleRace/TheHurdleRace.java
package algorithms.TheHurdleRace;

import java.util.Arrays;
import java.util.Scanner;

// https://www.hackerrank.com/challenges/the-hurdle-race/problem
public class TheHurdleRace {
    static int hurdleRace(int k, int[] height) {
        int max = Arrays.stream(height).max().getAsInt();
        return k - max >= 0 ? 0 : max - k;
    }

    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) {

        String[] nk = scanner.nextLine().split(" ");
        int n = Integer.parseInt(nk[0]);
        int k = Integer.parseInt(nk[1]);

        int[] height = new int[n];
        String[] heightItems = scanner.nextLine().split(" ");

        for (int i = 0; i < n; i++) {
            int heightItem = Integer.parseInt(heightItems[i]);
            height[i] = heightItem;
        }

        int result = hurdleRace(k, height);
        System.out.println(result);

        scanner.close();
    }
}

File: hackerRank/algorithms/TheHurdleRace/the-hurdle-race.py
# https://www.hackerrank.com/challenges/the-hurdle-race/problem
def hurdleRace(k, height):
    return max(max(height) - k, 0)


if __name__ == '__main__':
    nk = input().split()

    n = int(nk[0])
    k = int(nk[1])

    height = list(map(int, input().rstrip().split()))
    result = hurdleRace(k, height)
    print(result)